client/src/templates/Challenges/generic/show.tsx

```diff
@@ -0,0 +1,269 @@
+      helpCategory
+    });
+    challengeMounted(challengeMeta.id);
+  }, [
+    title,
+    challengeMeta,
+    challengeType,
+    helpCategory,
+    challengeMounted,
+    updateChallengeMeta
+  ]);
+
+  // video
+  const [videoIsLoaded, setVideoIsLoaded] = useState(false);
+
+  const handleVideoIsLoaded = () => {
+    setVideoIsLoaded(true);
+  };
+
+  // assignments
+  const [allAssignmentsCompleted, setAllAssignmentsCompleted] = useState(false);
+  const [assignmentsCompleted, setAssignmentsCompleted] = useState(0);
+
+  const handleAssignmentChange = (
+    event: React.ChangeEvent<HTMLInputElement>,
+    totalAssignments: number
+  ) => {
+    const newAssignmentsCompleted = event.target.checked
+      ? assignmentsCompleted + 1
+      : assignmentsCompleted - 1;
+
+    setAssignmentsCompleted(newAssignmentsCompleted);
+    setAllAssignmentsCompleted(totalAssignments === newAssignmentsCompleted);
+  };
+
+  // submit
+  const handleSubmit = () => {
+    if (assignments.length > 0 && allAssignmentsCompleted) {
+      openCompletionModal();
+    }
+  };
+
+  return (
+    <Hotkeys
+      executeChallenge={handleSubmit}
+      containerRef={container}
+      nextChallengePath={nextChallengePath}
+      prevChallengePath={prevChallengePath}
+    >
+      <LearnLayout>
+        <Helmet
+          title={`${blockNameTitle} | ${t('learn.learn')} | freeCodeCamp.org`}
+        />
+        <Container>
+          <Row>
+            <Spacer size='medium' />
+            <ChallengeTitle
+              isCompleted={isChallengeCompleted}
+              translationPending={translationPending}
+            >
+              {title}
+            </ChallengeTitle>
+
+            <Col md={8} mdOffset={2} sm={10} smOffset={1} xs={12}>
+              {description && (
+                <ChallengeDescription description={description} />
+              )}
+            </Col>
+
+            <Col lg={10} lgOffset={1} md={10} mdOffset={1}>
+              {videoId && (
+                <VideoPlayer
+                  bilibiliIds={bilibiliIds}
+                  onVideoLoad={handleVideoIsLoaded}
+                  title={title}
+                  videoId={videoId}
+                  videoIsLoaded={videoIsLoaded}
+                  videoLocaleIds={videoLocaleIds}
+                />
+              )}
+            </Col>
+
+            <Col md={8} mdOffset={2} sm={10} smOffset={1} xs={12}>
+              {instructions && (
+                <ChallengeDescription description={instructions} />
+              )}
+
+              <Spacer size='medium' />
+
+              {assignments.length > 0 && (
+                <Assignments
+                  assignments={assignments}
+                  allAssignmentsCompleted={allAssignmentsCompleted}
+                  handleAssignmentChange={handleAssignmentChange}
+                />
+              )}
+
+              <Button block={true} variant='primary' onClick={handleSubmit}>
+                {t('buttons.check-answer')}
```

Comment:
is there a way to conditionally update this text based on if it is a review page or not?

because for the review pages it won't make much sense to say check your answer

<img width="952" alt="Screenshot 2024-10-10 at 9 55 41 AM" src="https://github.com/user-attachments/assets/6ec1be48-d1c1-4870-a6e9-7ce8d9a1a3f8">

maybe, proceed to quiz instead?

Reply:
✅

Reply:
I change it to just "Submit" for now, does that work?

Reply:
Yeah, submit works well here 👍